File: src/Algorithms/PascalsTriangle.java
package Algorithms;

import java.util.ArrayList;
import java.util.List;

public class PascalsTriangle {
    public static List<List<Integer>> generate(int numRows) {

        List<List<Integer>> result = new ArrayList<>();
        List<Integer> previous = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> eachListItem = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                if (j == 0) {
                    eachListItem.add(1);
                    continue;
                }

                if (j == i) {
                    eachListItem.add(1);
                    continue;
                }
                eachListItem.add(previous.get(j) + previous.get(j - 1));
            }
            previous = eachListItem;
            result.add(eachListItem);

        }

        return result;

    }

    public static void main(String[] args) {

        System.out.println(generate(5));

    }
}

File: Python/Algorithms/PascalsTriangle.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        for i in range(0, numRows):
            internalResult = []
            for j in range(0, i + 1):
                if j == 0:
                    internalResult.append(1)
                    continue
                if j == i:
                    internalResult.append(1)
                    continue
                internalResult.append(previousList[j] + previousList[j - 1])
            previousList = internalResult
            result.append(internalResult)
        return result


if __name__ == '__main__':
    solution = Solution()
    res = solution.generate(5)
    print(res)
